Wrappers/Python/ccpi/viewer/CILViewer2D.py

```diff
@@ -1134,66 +1084,35 @@ def DisplayLineProfile(self, interactor, event, display):
 ###############################################################################
 
 
-class CILViewer2D():
+class CILViewer2D(CILViewerBase):
     '''Simple Interactive Viewer based on VTK classes'''
     # visualisation modes
     IMAGE_WITH_OVERLAY = 0
     RECTILINEAR_WIPE = 1
 
     def __init__(self, dimx=600, dimy=600, ren=None, renWin=None, iren=None, debug=True):
-        '''creates the rendering pipeline'''
-        # create a rendering window and renderer
-        if ren == None:
-            self.ren = vtk.vtkRenderer()
-        else:
-            self.ren = ren
-        if renWin == None:
-            self.renWin = vtk.vtkRenderWindow()
-        else:
-            self.renWin = renWin
-        if iren == None:
-            self.iren = vtk.vtkRenderWindowInteractor()
-        else:
-            self.iren = iren
-        # holder for list of actors
-        self.actors = {}
-        self.debug = debug
-
-        self.renWin.SetSize(dimx, dimy)
-        self.renWin.AddRenderer(self.ren)
+        CILViewerBase.__init__(self, dimx=600, dimy=600, ren=None, renWin=None, iren=None, debug=True)
 
         self.style = CILInteractorStyle(self)
-        self.style.debug = debug
-
         self.iren.SetInteractorStyle(self.style)
         self.iren.SetRenderWindow(self.renWin)
```

Comment:
I guess this line needs to be documented in the base class, because it is necessary to set the render window to the interactor and this is not done in the base class

Reply:
moved SetRenderWindow line to base class
Documented setting of interactor style in base class